src/libAtomVM/avmpack.h

```diff
@@ -25,12 +25,18 @@
 #ifndef _AVMPACK_H_
 #define _AVMPACK_H_
 
+#include "list.h"
 #include <stdint.h>
 
 #define END_OF_FILE 0
 #define BEAM_START_FLAG 1
 #define BEAM_CODE_FLAG 2
 
+typedef struct {
+    struct ListHead avmpack_head;
+    const void * data;
+} AVMPackData;
```

Comment:
As a general rule I'm trying to avoid introducing further typedefed structs, they are quite annoying to forward declare.
My long term plan here is to keep typedef just for Context and GlobalContext that are nearly used everywhere. Anyway feel free to share your opinion here :)

Reply:
I have no convictions, one way or another.  I have not had issues with typedefs in C++ in the past (boost is arguably one big typedef!), but C is not C++.